docs/src/reference/types.md

```diff
@@ -901,3 +901,73 @@ You can access definitions from the module using
 >>>
 >>> #(-3)
 ```
+
+# Selector
+A filter for selecting elements within the document.
+
+You can construct a selector in the following ways:
+- you can call the `[selector]` function
+- you can use a `[position]`
+- you can use an element function
+- you can use a regex
+- you can use a `[<label>]`
+
+A selector is what you can use to query the document for certain types
+of elements. It can also be used to apply styling rules to element. You can
+combined multiple queries together using the methods on a selector.
+
+Selectors can also be passed to several of the built-in functions in typst to
+change their behaviour. Such as the [outline]($func/outline) where it can be
+use to change the type of items that are listed within the outline.
+
+## Example
+```example
+#locate(loc => {
+  repr(query(heading.where(level: 1).any(heading.where(level: 2)), loc))
+})
+
+= This will be found by the query
+
+== So will this
+
+=== But this will not.
+
+```
+
+## Methods
+### before()
+Returns a new selector that will only match elements that were declared
+before the first match of the selector argument.
+
+- location: selector (positional, required)
+  The selector ued to find the last element.
```

Comment:
```suggestion
- target: selector (positional, required)
  The selector used to find the last element.
```
or maybe `end` instead of `target` and `start` for `after()`?
in any case, these suggestions also apply to `after()`.